trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import gpflux
+import gpflow
+import tensorflow as tf
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+from ..optimizer import Optimizer, TFOptimizer
+from ..gpflow.utils import module_deepcopy
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model. Note that Scipy optimizer
```

Comment:
note about Scipy is probably not needed here

Reply:
Done